sdk/test-utils/recorder/package.json

```diff
@@ -61,6 +61,7 @@
     "fs-extra": "^8.1.0",
     "nise": "^1.4.10",
     "nock": "^11.7.0",
+    "url-parse": "^1.4.7",
```

Comment:
why pull in a third party URL parser when we have our own? https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/core/core-http/lib/url.ts#L78

Or alternatively, if we want to move this to third-party code, should we get rid of our own?

Reply:
I'm good either way.

Reply:
IMO, we shouldn't have our own implementations of stuff like URL parsing when there's a good community standard, and url-parse is pretty much that, unless there's a really major benefit to having it in-tree

that said, we should make sure to test url-parse thoroughly on stuff like blob SAS urls before we ditch our own if we decide to go down that route. URLs can get pretty gnarly and we wouldn't want our stuff to break because we threw away a working implementation for a dependency.... For a little while, even [Chrome would choke on those SAS URLs](https://stackoverflow.com/questions/56636104/chrome-has-started-blocking-azure-blob-storage-downloads).

Reply:
Also @xirzec I think that function is only for parsing the query parameters part of a URL.

Reply:
No. URLBuilder from core-http allows parsing and getting the host, path, etc from a given URL.

Reply:
There's also https://nodejs.org/api/url.html#url_url_parse_urlstring_parsequerystring_slashesdenotehost , isn't that enough?

Reply:
`url-parse` version - af30705
`coreHttp.URLBuilder` version - 095059a